app/controllers/correctReturn/CorrectReturnCYAController.scala

```diff
@@ -19,31 +19,59 @@ package controllers.correctReturn
 import models.SelectChange.CorrectReturn
 import pages.correctReturn.CorrectReturnBaseCYAPage
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.ReturnService
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryList
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import utilities.GenericLogger
 import views.html.correctReturn.CorrectReturnCYAView
 import views.summary.correctReturn.CorrectReturnBaseCYASummary
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 class CorrectReturnCYAController @Inject()(override
                                            val messagesApi: MessagesApi,
                                            controllerActions: ControllerActions,
                                            val controllerComponents: MessagesControllerComponents,
                                            requiredUserAnswers: RequiredUserAnswersForCorrectReturn,
-                                           view: CorrectReturnCYAView)
-                                          (implicit config: FrontendAppConfig) extends FrontendBaseController with I18nSupport {
+                                           returnService: ReturnService,
+                                           view: CorrectReturnCYAView,
+                                           genericLogger: GenericLogger)
+                                          (implicit config: FrontendAppConfig, ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
   def onPageLoad: Action[AnyContent] = controllerActions.withCorrectReturnJourneyData.async {
     implicit request =>
       requiredUserAnswers.requireData(CorrectReturnBaseCYAPage) {
-        val orgName: String = " " + request.subscription.orgName
-        val sections = CorrectReturnBaseCYASummary.summaryListAndHeadings(request.userAnswers, request.subscription)
-        Future.successful(Ok(view(orgName, sections, routes.CorrectReturnCYAController.onSubmit)))
+        request.userAnswers.getCorrectReturnOriginalSDILReturnData.map(originalSdilReturn => {
+          val balanceBroughtForward = returnService.getBalanceBroughtForward(request.sdilEnrolment)
+          val orgName: String = " " + request.subscription.orgName
+
+          def sections(balanceBroughtForward: BigDecimal): Seq[(String, SummaryList)] = {
+            CorrectReturnBaseCYASummary.summaryListAndHeadings(
+              request.userAnswers,
+              request.subscription,
+              amounts = Amounts(
+                originalReturnTotal = originalSdilReturn.total,
+                newReturnTotal = SdilReturn(request.userAnswers).total,
+                balanceBroughtForward = balanceBroughtForward * -1,
+                adjustedAmount = if(balanceBroughtForward == 0){SdilReturn(request.userAnswers).total + balanceBroughtForward} else {
```

Comment:
This reduces to `adjustedAmount = SdilReturn(request.userAnswers).total + (balanceBroughtForward * -1)`